### PR TITLE
[RISCV] Account for zvfhmin and zvfbfmin promotion in register usage

### PR DESCRIPTION
**Pull request by lukel97**
A half with only zvfhmin or bfloat will end up getting promoted to a f32 for most instructions.

Unless the loop consists only of memory ops and permutation instructions which don't need promoted (is this common?), we'll end up using double the LMUL than what's currently being returned by getRegUsageForType.

Since this is used by the loop vectorizer, it seems better to be conservative and assume that any usage of a zvfhmin half/bfloat will end up being widened to a f32

### CODE REVIEW

**Review by topperc — APPROVED** [review]
Seems reasonable to me.

**Review by jacquesguan — APPROVED** [review]
LGTM

File: llvm/test/Transforms/LoopVectorize/RISCV/reg-usage-bf16.ll [hunk 1]
```diff
@@ -0,0 +1,31 @@
+; RUN: opt -passes=loop-vectorize -mtriple riscv64 -mattr=+v,+zvfbfmin -debug-only=loop-vectorize -riscv-v-register-bit-width-lmul=1 -S < %s 2>&1 | FileCheck %s
```
Comment (topperc) [inline llvm/test/Transforms/LoopVectorize/RISCV/reg-usage-bf16.ll:1]: Need `REQUIRES: asserts`

Comment (lukel97) [reply llvm/test/Transforms/LoopVectorize/RISCV/reg-usage-bf16.ll:1]: Done in 30d7dcc1db476b828d3b0a7b5c9bcfbe5bd5f216

**Pull request merged**